tests/api.test.js

```diff
@@ -2,123 +2,108 @@ import { describe } from "node:test";
-    expect(res.statusCode).toBe(200);
-    expect(res.body.name).toBe("JaneDoe");
-    expect(res.body.email).toBe("something@example.mail");
+  test("should try to get challenge with incorrect date and return status code 404", async () => {
+    const res = await request(URL).get("/api/v1/challenges/2021-01-01");
+    expect(res.statusCode).toBe(404);
   });
 
-  // test("DELETE /api/v1/users/1", async () => {
-  //   const res = await request(URL).delete("/api/v1/users/1");
-  //   expect(res.statusCode).toBe(204);
-  // });
+  test("should send bad request and return status code 400", async () => {
+    const res = await request(URL).get("/api/v1/challenges/foobar/");
+    expect(res.statusCode).toBe(400);
+  });
+
+  test("should request nonexistent challenge and return status code 404", async () => {
+    const res = await request(URL).get("/api/v1/challenges/1000");
+    expect(res.statusCode).toBe(404);
+  });
 });
 
-describe("CHALLENGES table tests", () => {
-  test("GET all /challenges", async () => {
-    const res = await request(URL).get("/api/v1/challenges/today");
+describe("COMPLETIONS SUITE", () => {
+  test("should get all completions and return status code 200", async () => {
+    const res = await request(URL).get("/api/v1/completions/1");
     expect(res.statusCode).toBe(200);
   });
 
-  test("GET from challenge with ID 1", async () => {
-    const res = await request(URL).get("/api/v1/challenges/1");
+  test("should send bad request and get status code 400", async () => {
+    const res = await request(URL).get("/api/v1/completions/foobar");
+    expect(res.statusCode).toBe(400);
+  });
+
+  test("should request user stats and return status code 200", async () => {
+    const res = await request(URL).get("/api/v1/completions/stats");
     expect(res.statusCode).toBe(200);
   });
 
-  test("POST to /challenges", async () => {
-    const res = await request(URL)
-      .post("/api/v1/challenges")
+  test("should send bad request of completion based on id and return status code 400", async () => {
+    const res = await request(URL).get("/api/v1/completions/foobar");
+    expect(res.statusCode).toBe(400);
+  });
+
+  test("should request nonexistent completion and return status code 404", async () => {
+    const res = await request(URL).get("/api/v1/completions/1000");
+    expect(res.statusCode).toBe(404);
+  });
+
+  test("should post a new completion and return status code 201", async () => {
+    const challengeId = 11;
+    const userId = 11;
+    const description = "Completion 11 description";
+
+    const response = await request("/api/v1/completions/")
+      .post("/api/v1/completions")
       .send({
-        prompt: "Do something kind",
-        description: "Do something kind for someone else",
-        suggestion: "Buy someone a coffee",
-        date: "2021-10-01",
-        source: "ChatGPT",
-        image: "https://example.com/image.jpg",
+        challenge_id: challengeId,
```

Comment:
The challenge is based off the day the post request happens in the backend, but adding this line wont make the test fail.

Reply:
For this test, how should I be tested and how should the data get defined, assuming that it needs to be?

Reply:
I just made a bunch of changes to wherever a user is used to integrate kinde. On the request object, I think you need to attach an object called user that has an attribute id (you'd access it req.user.id if that helps) then in the body of the request you add the description. But the backend handles pretty much everything else. If you're trying to make a completion that already exists you are redirected to the existing completion -- so be aware of that for testing purposes. Like if you run the tests twice in a row, I think the second test would fail.